specification.md

```diff
@@ -0,0 +1,344 @@
+        purpose, schema, or versioning. If it were just "application/json", for
+        example, then every application would need to impose some "type" field
+        within the payload, lest we have similar vulnerabilities as if
+        payloadType were not signed.
+    -   Also, URIs don't need to be registered while Media Types do.
+
+-   Why use payloadType "backwards-compatible-json" instead of assuming
+    backwards compatible mode if payloadType is absent?
+
+    -   We wanted to leave open the possibility of having an
+        application-specific "default" value if payloadType is unspecified,
+        rather than forcing the default to be backwards compatibility mode.
+    -   Note that specific applications can still choose backwards compatibility
+        to be the default.
+
+-   Why not stay backwards compatible by requiring the payload to always be JSON
+    with a "_type" field? Then if you want a non-JSON payload, you could simply
+    have a field that contains the real payload, e.g. `{"_type":"my-thing",
+    "value":"base64…"}`.
+
+    1.  It encourages users to add a "_type" field to their payload, which in
+        turn:
+        -   (a) Ties the payload type to the authentication type. Ideally the
+            two would be independent.
+        -   (b) May conflict with other uses of that same field.
+        -   (c) May require the user to specify type multiple times with
+            different field names, e.g. with "@context" for
+            [JSON-LD](https://json-ld.org/).
+    2.  It would incur double base64 encoding overhead for non-JSON payloads.
+    3.  It is more complex than PAE.
+
+# Backwards Compatibility
+
+## Current format
+
+The
+[current signature format](https://github.com/in-toto/docs/blob/master/in-toto-spec.md#42-file-formats-general-principles)
+used by TUF and in-toto has a BODY that is a regular JSON object and a signature over the
+[Canonical JSON] serialization of BODY.
+
+```json
+{
+  "signed": <BODY>,
+  "signatures": [{
+    …,
+    "sig": "<Hex(Sign(CanonicalJson(BODY)))>"
+  }, …]
+}
+```
+
+To verify, the consumer parses the whole JSON file, re-serializes BODY using
+Canonical JSON, then verifies the signature.
+
+## Detect if a document is using old format
+
+To detect whether a signature is in the old or new format:
+
+-   If it contains a `payload` field, assume it is in the new format.
+-   If it contains a `signed` field, assume it is in the old format.
+
+To convert an existing signature to the new format:
+
+-   `new.payload = base64encode(CanonicalJson(orig.signed))`
+-   `new.payloadType = "<URI>/backwards-compatible-json"`
+-   `new.signatures[*].sig = base64encode(hexdecode(orig.signatures[*].sig))`
+
+To convert a backwards compatible signature to the old format:
+
+-   `old.signed = jsonparse(base64decode(new.payload))`
+-   `old.signatures[*].sig = hexencode(base64decode(new.signatures[*].sig))`
+
+# Security
+
```

Comment:
Is this section required? Even if empty?

Reply:
Good point, it's a necessary part of an ITE. I'll get rid of it for now, and we can add it back in if there's content.